golos.referral/src/golos.referral.cpp

```diff
@@ -63,23 +82,23 @@ void referral::addreferral(name referrer, name referral, uint32_t percent,
     });
 }
 
-void referral::transfer(name referral, asset quantity) {
-    require_auth(_self);
-    asset balance = eosio::token::get_balance(config::token_name, _self, quantity.symbol.code());
-    eosio_assert(balance.amount >= quantity.amount, "Insufficient funds in the gls.referral account.");
+void referral::on_transfer(name from, name to, asset quantity, std::string memo) {
+    if(_self != to)
+        return;
+
+    if(token::get_issuer(config::token_name, quantity.symbol.code()) == from && to == name())
```

Comment:
`to == _self` at this point (see 1st `if` at the top of function), so condition of this `if` will always be `false`